docs/editor-neovim.qmd

```diff
@@ -0,0 +1,59 @@
+---
+title: "Neovim"
+editor:
+  markdown:
+    wrap: sentence
+    canonical: true
+---
+
+# Installing the Air server
+
+First, install Air by following the instructions on the [Air CLI documentation](https://posit-dev.github.io/air/cli.html). The CLI binary bundles the Air server as well as a command line formatting tool.
+
+# Configuration
+
+Air can be configured as a Language Server Protocol (LSP) server via [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) or as a formatter plugin via [conform.nvim](https://github.com/stevearc/conform.nvim).
+
+## LSP Configuration with nvim-lspconfig
+
+To configure the client side to use Air's language server, add the following to your configuration. This ensures that Air automatically formats R code when you save a file:
+
+```lua
+require("lspconfig").air.setup({
+    on_attach = function(_, bufnr)
+        vim.api.nvim_create_autocmd("BufWritePre", {
+            buffer = bufnr,
+            callback = function()
+                vim.lsp.buf.format()
+            end,
+        })
+    end,
+})
```

Comment:
Can someone help me understand why the `on_attach` is required?

I'm comparing to ruff, for example:
https://docs.astral.sh/ruff/editors/setup/#neovim

That does not have this `on_attach` hook to enable the `BufWritePre` pre-write hook. I would have thought that just setting up Air as an LSP would have activated something like this on its own.

Is Ruff's documentation missing something? Is this `BufWritePre` overkill? Or am I missing something else?

Reply:
Technically, it's not reuiqred. However, without it formatting will work only on demand (by manually running `vim.lsp.buf.format()`). Adding this particular `on_attach` effectively enables formatting on save.

Reply:
Ok so IIUC its like `editor.formatOnSave` in VS Code. Without that, you can invoke `Format Document` to tell the LSP to format your document, but with that setting set to `true` it will do so automatically on every save.

I guess in ruff they just didn't think to say that part.

But makes sense to me thanks!